snuba/replacers/replacer_processor.py

```diff
@@ -0,0 +1,67 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Any, Mapping, NamedTuple, Optional
+
+from snuba.datasets.schemas.tables import TableSchema, WritableTableSchema
+
+
+class ReplacementMessage(NamedTuple):
```

Comment:
I was considering making the ReplacerProcessor Generic with regards to the ReplacementMessage type, but I did not do that for two reasons:
1) ReplacementMessage represents what we get from the Kafka topic, so it is not really generic
2) if it was generic we would still need a way for this class to instantiate the proper replacer specific object, which reduces a lot the advantage of the generic.